reptyr: enable on 32-bit ARM Linux and x86 FreeBSD

lopsided98 opened the pull request:
###### Motivation for this change

Reptyr should not restricted to specific architectures. I tested it and it works fine on armv7l.

###### Things done

Made the platforms less restrictive, allowing it to run on all Linux architectures. MacOS is not supported.

cc @7c6f434c

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

lopsided98 committed:
reptyr: enable on 32-bit ARM Linux and x86 FreeBSD

thoughtpolice commented:
@GrahamcOfBorg build reptyr

GrahamcOfBorg commented:
<!--REQUEST_ID=072661ae-0be9-412a-9ee6-07a3d3c5fb8f-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=0969a6db-3ca1-441c-ad48-5a60d2408233)

Attempted: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (0.02 MiB download, 0.05 MiB unpacked):
  /nix/store/2dznhkbbrjjcppm6fxwh6n8h5ia5kpzw-reptyr-0.6.2
copying path '/nix/store/2dznhkbbrjjcppm6fxwh6n8h5ia5kpzw-reptyr-0.6.2' from 'https://cache.nixos.org'...
/nix/store/2dznhkbbrjjcppm6fxwh6n8h5ia5kpzw-reptyr-0.6.2
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=072661ae-0be9-412a-9ee6-07a3d3c5fb8f-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=feb45ec0-b29c-48ec-ae81-c77dca1f2379)

Attempted: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```
build flags: SHELL=/nix/store/bpspdmsl0yys24gs70flsvcw3wcnl7lx-bash-4.4-p23/bin/bash PREFIX=\$\(out\)
gcc -Wall -Werror -D_GNU_SOURCE -g   -c -o reptyr.o reptyr.c
In file included from platform/platform.h:28:0,
                 from reptyr.c:37:
platform/../ptrace.h:66:14: error: field 'user' has incomplete type
  struct user user;
              ^~~~
make: *** [<builtin>: reptyr.o] Error 1
builder for '/nix/store/4lx8qfb8sik1y32dspanm90sv9ahrhcj-reptyr-0.6.2.drv' failed with exit code 2
error: build of '/nix/store/4lx8qfb8sik1y32dspanm90sv9ahrhcj-reptyr-0.6.2.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=072661ae-0be9-412a-9ee6-07a3d3c5fb8f-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=dda9106a-9ec1-40b7-b276-b79dc589b85a)

The following builds were skipped because they don't evaluate on x86_64-darwin: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

lopsided98 commented:
Oops, didn't test this on aarch64. I guess it only supports 32-bit ARM.

lopsided98 commented:
I updated the platforms to support 32-bit ARM, and added freebsd as well because it is supposedly supported, although I don't have a machine to test it with.

I also updated the description to the current GitHub description, which reflects that fact that it is no longer only supported on Linux.

7c6f434c commented:
Would `with stdenv.lib.platforms; linux ++ freebsd` work then?

c0bw3b commented:
`platforms.linux` would include aarch64-linux and it seems it fails to build on ARMv8

c0bw3b commented on pkgs/os-specific/linux/reptyr/default.nix:
```diff
@@ -20,9 +20,14 @@ stdenv.mkDerivation rec {
 
   makeFlags = ["PREFIX=$(out)"];
   meta = {
```
Also add `meta.homepage`

lopsided98 replied:
Done.

lopsided98 committed:
reptyr: add homepage

thoughtpolice commented:
@GrahamcOfBorg build reptyr

GrahamcOfBorg commented:
<!--REQUEST_ID=d93cf681-efc2-4da7-9db9-6a8378847e07-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=a1b58e46-0566-4165-99c4-4e24a0fab008)

The following builds were skipped because they don't evaluate on aarch64-linux: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=d93cf681-efc2-4da7-9db9-6a8378847e07-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=322b9fce-8508-4937-bc64-9aef64e9777f)

The following builds were skipped because they don't evaluate on x86_64-darwin: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=d93cf681-efc2-4da7-9db9-6a8378847e07-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51259&attempt_id=fae7c9b2-498a-4305-9078-aee73a94fedc)

Attempted: reptyr

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (0.02 MiB download, 0.05 MiB unpacked):
  /nix/store/la1lrgd8lq5jfxfmxy8fxfzcq1jwgbyj-reptyr-0.6.2
copying path '/nix/store/la1lrgd8lq5jfxfmxy8fxfzcq1jwgbyj-reptyr-0.6.2' from 'https://cache.nixos.org'...
/nix/store/la1lrgd8lq5jfxfmxy8fxfzcq1jwgbyj-reptyr-0.6.2
```
</p></details>
